### PR TITLE
Use SDL_MOUSE_TOUCH_EVENTS in touch test and docs

### LINKED ISSUE #1182: touchpad not giving FINGER events on MacOS Catalina SDL 2.0.10 and above (2309)
#  **Issue &numero;2309 opened by *[illume](https://github.com/illume)* at 2020-11-07 08:26:33**

Edit: SDL_MOUSE_TOUCH_EVENTS=1 is required to get FINGER and touch events from the touchpad from SDL 2.0.10 onwards.

- [ ] tests/touch_test.py needs some refactoring. pygame.init() is required for `touch.get_num_devices()` to return non zero (at least on mac with SDL_MOUSE_TOUCH_EVENTS=1 and a touchpad).
- [ ] to make testing this easier with tests/touch_test.py, setting SDL_MOUSE_TOUCH_EVENTS=1 in the test could be a good idea?
- [ ] SDL_MOUSE_TOUCH_EVENTS=1 should probably be documented in the touch docs (and maybe in the event docs where FINGER_ are).

---
The new interactive test in https://github.com/pygame/pygame/pull/2279 shows FINGER events are not working.

I have a wacom touch device, and a macbook touch pad I tried on MacOS.

`pygame._sdl2.touch.get_num_devices()` is returning 0.

- I tried the SDL-2.0.12 test/testgesture and it wasn't working with either, so I'm sure it's not a pygame issue.
- Then I tried pygame 2.0.0.dev6 (SDL 2.0.10, python 2.7.16) and it didn't work either. Which makes me start to think it's to do with the Mac version I have... MacOS Catalina 10.15.7. In a previous version of MacOS it was working for me on this same hardware... ?
- Then I tried with `pygame==2.0.0.dev1` and it worked! So it could be SDL 2.0.10 which introduced the issue. Or the way it was compiled. 
```
python -m pygame.examples.eventlist
pygame 2.0.0.dev1 (SDL 2.0.9, python 2.7.16)
```
- I tried my locally compiled pygame with homebrew SDL 2.0.12, and it didn't work.
- I tried pygame 2 with conan compiled SDL 2.0.12, and it didn't work.

<hr>

#  Comments

# # #  **[illume](https://github.com/illume)* commented at 2020-11-09 09:27:13*

Tracked this down to: https://bugzilla.libsdl.org/show_bug.cgi?id=5152

> Sam Lantinga 2020-05-29 23:36:39 UTC
>
>We changed it so Mac trackpad events come in as mouse events by default, to match most customers' expectations.
>
>You can change this by setting SDL_HINT_MOUSE_TOUCH_EVENTS to "1", or as an environment variable SDL_MOUSE_TOUCH_EVENTS=1 before you run your game, to verify that this fixes it.
>
> Feel free to reopen this bug if that's not the problem here.

To see finger and gesture events with a touchpad do: 
```
SDL_MOUSE_TOUCH_EVENTS=1 python3 -m pygame.examples.eventlist
```

### PR DESCRIPTION
**Pull request by MyreMylar**
fixes #1182

### CODE REVIEW

File: test/touch_test.py [hunk 1]
```diff
@@ -40,13 +41,24 @@ def test_num_fingers__invalid(self):
...
+        pygame.quit()
+
     def test_get_finger(self):
         """ask for touch input and check the dict"""
 
-        pygame.display.init()
-        pygame.font.init()
-
         os.environ["SDL_VIDEO_WINDOW_POS"] = "50,50"
         screen = pygame.display.set_mode((800, 600))
```
Comment (ankith26) [inline test/touch_test.py:63]:
```suggestion
        screen = pygame.display.set_mode((1000, 600))
```
minor update, 800 is too small for the event text being displayed

**Review by ankith26 — APPROVED** [review]
LGTM, left a minor related change suggestion for an issue observed while testing locally.

File: docs/reST/ref/pygame.rst [hunk 2]
```diff
@@ -535,3 +535,13 @@ By default, when the window is not in focus, input devices do not get
 updated. However, using this environment variable it is possible to get
 joystick updates even when the window is in the background. Must be set
 before calling :func:`pygame.init()` or :func:`pygame.joystick.init()`.
+
```
Comment (ankith26) [inline docs/reST/ref/pygame.rst:538]:
```suggestion

|

```

File: test/touch_test.py [hunk 3]
```diff
@@ -5,32 +5,33 @@
...
         self.assertRaises(TypeError, touch.get_device, "test")
 
-    @unittest.skipIf(not has_touchdevice, "no touch devices found")
     def test_num_fingers(self):
-        touch.get_num_fingers(touch.get_device(0))
+        if touch.get_num_devices() > 0:
+            if touch.get_device(0) >= 0:
+                # mice acting as touch devices have negative device IDs
+                # they also do not work with get_num_fingers()
+                touch.get_num_fingers()
```
Comment (ankith26) [inline test/touch_test.py:34]:
```suggestion
                touch.get_num_fingers(touch.get_device(0))
```

**Review by zoldalma999 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**